internal/fs/inode/dir.go

```diff
@@ -790,6 +819,75 @@ func (d *dirInode) ReadEntries(
 	return
 }
 
+// HasNoSupportedObjectsInSubtree returns true if the GCS prefix corresponding
+// to this directory contains only objects of the form `<a>//<b>` where <a> and
+// <b> are proper path strings not containing a `//`.
+// If there are no GCS objects for this prefix, then it will return false.
+// Example: Input directory inode: "a/".
+// If there is even one GCS object like "a/b" or "a/b/c" etc, then it will return false.
+// If it contains only "a//b" or "a/b//c" or "a//b/c" etc. then it will return true.
+// Note: This is a recursive method.
+func (d *dirInode) HasNoSupportedObjectsInSubtree(ctx context.Context) (hasNoSupportedObjects bool, err error) {
+	if d.isBucketHierarchical() {
+		d.includeFoldersAsPrefixes = true
+	}
+
+	// Do a BFS traversal of the GCS prefix corresponding to this directory,
+	// to find out if it has any supported GCS objects
+	// (i.e. GCS objects not containing // in its name) at all.
+	bucket := d.bucket
+	// Go does not have a queue structure in it, so using a slice in its place.
+	var dirNameBfsQueue []string
+	dirNameBfsQueue = append(dirNameBfsQueue, d.Name().GcsObjectName())
+
+	for len(dirNameBfsQueue) != 0 { // if queue is not empty
```

Comment:
Do we really need BFS here? Why can't we just traverse on list response with prefix dirInodeObjectName + "/" and see if any object does not have "//"?

Reply:
let's discuss this offline.